72_config_and_simplify_tls.md

```diff
@@ -0,0 +1,426 @@
+    * `zmq_ip`: Listing IP of the revocation notifier
+    * `zmq_port`: Listing port of the revocation notifier
+    * `webhook_url`: URL to send notifications to
+  * `max_upload_size`: Upload size for policies to the verifier
+  * `measured_boot_policy_name`: Name of the policy that is used for measured
+    boot
+  * `severity_labels`: List of used severity levels
+  * `severity_policy`: List of regexes that match a event ID to a severity label
+  * `ignore_tomtou_errors`: If "time of measure, time of use" (ToMToU) errors
+    should be treated as a failure. (Should this part of the IMA policy instead
+    the Keylime configuration?)
+* `tenant.conf`: C
+  * `verifier_ip`: IP of the verifier
+  * `verifier_port`: Port of the verifier
+  * `registrar_ip`: IP of the registrar
+  * `registrar_port`: Port of the registrar
+  * `tls_dir`: See TLS simplification
+  * `enable_agent_mtls`:  See TLS simplification
+  * `client_key`: See TLS simplification
+  * `client_cert`: See TLS simplification
+  * `trusted_server_ca`: See TLS simplification
+  * `tpm_cert_store`: Location of the directory with the trusted EK CA certificates
+  * `max_payload_size`: Maximal size of the payload to send. Should match tmpfs size for the agent secure mount.
+  * `accept_tpm_hash_algs`: List of allowed hash algorithms
+    * Default: sha512,sha384,sha256,sha1
+  * `accept_tpm_encryption_algs`: List of accepted encryption algorithms. (Note: some parts of Keylime only support RSA)
+    * Default: ecc, rsa
+  * `accept_tpm_signing_algs`: List of supported signing algorithms
+    * Default: ecschnorr,rsassa
+  * `exponential_backoff`: Use exponential backoff for retries
+  * `retry_interval`: Time to retry again or base for exponential retry if enabled
+  * `max_retries`: Maximum number of retries
+  * `require_ek_cert`: Require EK certificate validation
+  * `ek_check_script`: Path to a script for validating the EK certificate. The following environment variables are set:
+    * EK: PEM encoded version of the EK
+    * EK_CERT: DER encoded EK certificate if available
+* `registrar.conf`: Registrar specific configurations
+  * `registrar_ip`: Listing IP of the registrar
+  * `registrar_port`: Listing port of the registrar (agents)
+  * `registrar_tls_port` Listing port of the registrar (server side)
+  * `tls_dir`: See TLS simplification
+  * `server_key`: See TLS simplification
+  * `server_cert`: See TLS simplification
+  * `trusted_client_ca`: See TLS simplification
+  * `database_url`: Database configuration
+  * `database_pool_sz_ovfl`: SQAlchemy setting
+  * `auto_migrate_db`: Automatically update DB schema using alembic
+* `ca.conf`: Configuration for the CA
+  * `cert_country`
+  * `cert_ca_name`
+  * `cert_state`
+  * `cert_locality`
+  * `cert_organization`
+  * `cert_org_unit`
+  * `cert_ca_lifetime`
+  * `cert_lifetime`
+  * `cert_bits`
+  * `cert_crl_dist`
+* `webapp.conf`: WebApp specific configuration. TODO: do we want to deprecate the webapp?
```

Comment:
@mpeters @THS-on @lkatalin @maugustosilva @aplanas Comments on this?

Reply:
This is an opportunity to clean the codebase, and archive the webapp. Maybe can be revamped later.

Reply:
I'm for removing it. I created a PR that does that: https://github.com/keylime/keylime/pull/1032

I think the best way to implement a new WebApp is to directly use the REST APIs not going through the tenant python code, which is kinda hacky.